services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/credentials/AbstractCredentialsManagementService.java

```diff
@@ -91,4 +122,51 @@ public void setTenantInformationService(final TenantInformationService tenantInf
                         : processReadCredentials(DeviceKey.from(result.getPayload(), deviceId), span));
     }
 
+    private Future<List<CommonCredential>> verifyAndEncodePasswords(final List<CommonCredential> credentials) {
+        // Check if we need to encode passwords
+        if (!needToEncode(credentials)) {
```

Comment:
personally, I find it easier to read (and understand) the *positive* condition, i.e.
```
if (needToEncode(credentials)) {
} else {
}
```